### PR TITLE
[bsp/stm32] add support for NUCLEO-F410RB board;

### PR DESCRIPTION
**Pull request by tanglj86**
## 拉取/合并请求描述：(PR description)

[
add support for NUCLEO-F410RB board; 
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by tanglj86** [comment]
该BSP仅支持UART和PIN驱动，支持MDK5生成工程。目前测试正常。

File: bsp/stm32/stm32f410-st-nucleo/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+1. 在 BSP 下打开 env 工具。
+
+2. 输入 `menuconfig` 命令配置工程，配置好之后保存退出。
+
+3. 输入 `pkgs --update` 命令更新软件包。
+
+4. 输入 `scons --target=mdk4/mdk5/iar` 命令重新生成工程。
+
+本章节更多详细的介绍请参考 [STM32 系列 BSP 外设驱动使用教程](../docs/STM32 系列 BSP 外设驱动使用教程. md)。
```
Comment (SummerLife) [inline bsp/stm32/stm32f410-st-nucleo/README.md]: 无法显示

File: bsp/stm32/stm32f410-st-nucleo/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+通过阅读快速上手章节开发者可以快速地上手该 BSP，将 RT-Thread 运行在开发板上。在进阶使用指南章节，将会介绍更多高级功能，帮助开发者利用 RT-Thread 驱动更多板载资源。
+
+## 开发板介绍
+
+探索者 STM32F410-Nucleo-64 是意法半导体推出的一款基于 ARM Cortex-M4 内核的开发板，最高主频为 84Mhz，该开发板具有丰富的板载资源，可以充分发挥 STM32F410RB 的芯片性能。
+
+开发板外观如下图所示：
+
+![board](figures/board.png)
```
Comment (SummerLife) [inline bsp/stm32/stm32f410-st-nucleo/README.md]: 请仔细阅读 readme，看看是否有无法显示的图片和显示错误的行。

File: bsp/stm32/stm32f410-st-nucleo/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+该开发板常用 ** 板载资源 ** 如下：
```
Comment (SummerLife) [inline bsp/stm32/stm32f410-st-nucleo/README.md]: 多空格

**Comment by SummerLife** [comment]
感谢你的提交，不过有一些细节问题还需要修改。

**Comment by tanglj86** [comment]
感谢大神的指点

**Comment by SummerLife** [comment]
应该没有问题了，请参考 https://github.com/RT-Thread/rt-thread/blob/9c544e069cfb13eeb1905cbcb0b913f8fa7e6dff/.travis.yml#L90 来添加该 BSP 的 CI 测试，通过后就可以合并了。@tanglj86

**Pull request merged**